controllers/activemqartemisservice_test.go

```diff
@@ -0,0 +1,484 @@
+						Kind:       "Service",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "debug",
+						Namespace: defaultNamespace,
+					},
+					Spec: corev1.ServiceSpec{
+						Type: corev1.ServiceTypeNodePort,
+						Selector: map[string]string{
+							"ActiveMQArtemis": crd.Name,
+						},
+						Ports: []corev1.ServicePort{
+							{
+								Port: 5005,
+							},
+						},
+					},
+				}
+				Expect(k8sClient.Create(ctx, debugService)).Should(Succeed())
+			}
+
+			serviceKey := types.NamespacedName{Name: crd.Name, Namespace: crd.Namespace}
+			createdCrd := &brokerv1beta1.ActiveMQArtemisService{}
+
+			By("Checking ready cr status")
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, serviceKey, createdCrd)).Should(Succeed())
+
+				if verbose {
+					fmt.Printf("Service STATUS: %v\n\n", createdCrd.Status.Conditions)
+				}
+				g.Expect(meta.IsStatusConditionTrue(createdCrd.Status.Conditions, brokerv1beta1.ReadyConditionType)).Should(BeTrue())
+
+			}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+			By("checking broker cr status insync - should be in the service if important to user")
+			brokerKey := types.NamespacedName{Name: crd.Name, Namespace: crd.Namespace}
+			brokerCrd := &brokerv1beta1.ActiveMQArtemis{}
+
+			var appPropsRv = ""
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, brokerKey, brokerCrd)).Should(Succeed())
+
+				if verbose {
+					fmt.Printf("Broker STATUS: %v\n\n", brokerCrd.Status)
+				}
+
+				condition := meta.FindStatusCondition(brokerCrd.Status.Conditions, brokerv1beta1.ConfigAppliedConditionType)
+				g.Expect(condition).NotTo(BeNil())
+
+				for _, externalConfig := range brokerCrd.Status.ExternalConfigs {
+					if externalConfig.Name == AppPropertiesSecretName(brokerCrd.Name) {
+						appPropsRv = externalConfig.ResourceVersion
+					}
+				}
+				g.Expect(appPropsRv).ShouldNot(BeEmpty())
+
+			}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+			By("deploying a matching app")
+			appName := "first-app" // a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')",
+			app := brokerv1beta1.ActiveMQArtemisApp{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ActiveMQArtemisApp",
+					APIVersion: brokerv1beta1.GroupVersion.Identifier(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      appName,
+					Namespace: defaultNamespace,
+				},
+				Spec: brokerv1beta1.ActiveMQArtemisAppSpec{
+
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"forWorkQueue": "true",
+						}},
+
+					Auth: []brokerv1beta1.AppAuthType{
+						brokerv1beta1.MTLS,
+					},
+
+					Capabilities: []brokerv1beta1.AppCapabilityType{
+						{
+							// Role: appName, TODO respect role for mtls
```

Comment:
I have a question regarding this TODO.
Indeed working on the tutorial, I noticed that I was setting commonNames in the certificates but these names aren't propagated to other fields anywhere.
So my question is, how does that work?

Reply:
the app operator looks for a secret and extracts the common name from the cert and uses that as the authenticated id for the cert login module, that id gets the associated the relevant producer and consumer roles. But there is currently only support for a single identity, the one from the single secret. It is just a limitation of the code at the moment that it does not support a different role.